posts/Yoann/2016-09-30-introducing-hibernate-search-sort-dsl.adoc

```diff
@@ -0,0 +1,88 @@
+----
+  .sort()
+    .byField("title")
+    .andByIndexOrder()
+  .createSort()
+----
+
+That way, if there are two books with the same title in your index, they will always keep the same relative order from one query to another.
+
+The score sort is also available with `byScore()`. Obviously, there's one key difference with that one: the sort is descending by default. Getting the least relevant entities first wouldn't be very useful.
+
+Last but not least, if your entity has some https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/?v=5.5#spatial[spatial fields] you may also build spatial sorts:
+
+[source, JAVA]
+----
+  .sort()
+    .byDistance()
+      .onField("location")
+      .fromLatitude(24.0d)
+      .andLongitude(32.0d)
+    .createSort()
+----
+ 
+== Advanced use
+
+The API also exposes some useful advanced features, so be sure to check the documentation when it's out. Or, you know, since it's a fluid API, you can simply use your IDE autocomplete feature and see what's available!
+
+Among the useful features, the missing values behavior. What if I'm sorting books by publishing date, and some of them haven't even been published yet?
+
+[source, JAVA]
+----
+  .sort()
+    .byField("publishingDate_sort").desc() // Most recently published first 
+      .onMissingValue().sortFirst() // Not published yet => put this upper on the list
+    .andByField("custom_id_sort") // Default for the case when multiple books have no publishing date
+  .createSort()
+----
+
+And of course, if you're using Elasticsearch, you may want to take advantage of that brand-new feature that appeared in the last snapshot, the one you just spotted this morning and that would really save you of a lot of trouble on your project. But, hey, the Hibernate Search team is not on the same time zone, and even if they're providing fast support, you're not getting the feature pushed into Hibernate Search in time to meet your deadline. Which is this evening, by the way.
+
+Well, guess what: you can use that feature anyway. The sorting API also allows using native sorts by passing the JSON description of this sort, which will be added to the Elasticsearch query as is:
+
+[source, JAVA]
+----
+  .sort()
+    .byNative("authors.name", "{'order':'asc', 'mode': 'min'}")
+    .andByField("title")
+  .createSort()
+----
+
+== Next...
+
+Of course, one could point out that it's not really backend-independent. The API itself mostly is, but the returned type (`Sort`) is clearly bound to Apache Lucene.
+
+Well, one day at a time: the API in its current form *can* be adapted the be completely agnostic, so it's paving the way to http://hibernate.org/search/roadmap/#hibernate-search-6-x[Hibernate Search 6.x], while still requiring no change to any other contract such as `FullTextQuery.setSort(Sort)`. And that means it will available directly in 5.6.0.Beta3!
+
+So be sure to check it out, and feel free to contact us for any question, problem or simply for feedback!
```

Comment:
Yeah, not sure it's correct English, but it felt more natural... You're right, better be safe.